### PR TITLE
Moving signature creation logic to a seperate class

### PR DESCRIPTION
**Pull request by ChaithanyaGK**
Signed-off-by: Chaithanya Ganta <ganta@adobe.com>

As part of fixing #169, moving signature creation logic to AwsSignature class.

### CODE REVIEW

File: src/main/java/org/gaul/s3proxy/AwsSignature.java [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+import com.google.common.collect.SortedSetMultimap;
+import com.google.common.collect.TreeMultimap;
+import com.google.common.io.BaseEncoding;
+import com.google.common.net.HttpHeaders;
+import com.google.common.net.PercentEscaper;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class AwsSignature {
```
Comment (andrewgaul) [inline src/main/java/org/gaul/s3proxy/AwsSignature.java]: Please reduce public visibility to private or package access.  Repeated below.

Comment (ChaithanyaGK) [inline src/main/java/org/gaul/s3proxy/AwsSignature.java]: Done.

**Comment by andrewgaul** [comment]
Thank you for contribution @ChaithanyaGK!

**Comment by andrewgaul** [comment]
The added file had Windows line endings which I corrected.  You might want to check your local configuration.  Unfortunately Checkstyle does not yet support flagging Windows line endings checkstyle/checkstyle#893.

**Comment by ChaithanyaGK** [comment]
Thanks @andrewgaul, will check my local configuration

**Pull request merged**